tools/run_tests/run_xds_tests.py

```diff
@@ -947,12 +1070,41 @@ def resize_instance_group(gcp,
 
 
 def patch_url_map_backend_service(gcp, backend_service):
+    '''change url_map's backend service'''
     config = {
         'defaultService':
             backend_service.url,
         'pathMatchers': [{
             'name': _PATH_MATCHER_NAME,
             'defaultService': backend_service.url,
+            'defaultRouteAction': None,
+        }]
+    }
+    logger.debug('Sending GCP request with body=%s', config)
+    result = gcp.compute.urlMaps().patch(
+        project=gcp.project, urlMap=gcp.url_map.name,
+        body=config).execute(num_retries=_GCP_API_RETRIES)
+    wait_for_global_operation(gcp, result['name'])
+
+
+def patch_url_map_weighted_backend_services(gcp, servicesWithWeights):
```

Comment:
Why make a separate function instead of adding an optional servicesWithWeights arg to `patch_url_map_backend_service`?

Reply:
Merged into one function